### PR TITLE
[20503] Clarify history depth vs max_samples_per_instance inconsistency

### PR DESCRIPTION
**Pull request by EduPonz**
This PR adds documentation for:

* eProsima/Fast-DDS#4417

@Mergifyio backport 2.12.x 2.10.x 2.6.x

### CODE REVIEW

**Comment by JesusPoderoso** [comment]
@richiprosima please test docs

File: docs/fastdds/dds_layer/core/policy/standardQosPolicies.rst [hunk 1]
```diff
@@ -478,24 +478,26 @@ The HistoryQos must be set consistently with the :ref:`resourcelimitsqospolicy`,
...
-  configured as |KEEP_ALL_HISTORY_QOS-api|, the resource limits are reached, and based on the :ref:`durabilityqospolicy`
-  configuration:
-
-  * If the |DurabilityQosPolicy::kind-api| is configured as |VOLATILE_DURABILITY_QOS-api|, the DataWriter publication
-    will discard the oldest samples.
-    Those samples may belong to different :ref:`instances<dds_layer_topic_instances>`.
-  * If the |DurabilityQosPolicy::kind-api| is configured as |TRANSIENT_LOCAL_DURABILITY_QOS-api| or
-    |TRANSIENT_DURABILITY_QOS-api|, the DataWriter publication will be blocked.
+* In the case of the :ref:`reliabilityqospolicy` being set to |RELIABLE_RELIABILITY_QOS-api| and the
+  :ref:`historyqospolicy` being set to |HistoryQosPolicy::kind-api|, when the resource limits are reached, the behavior
```
Comment (JesusPoderoso) [inline docs/fastdds/dds_layer/core/policy/standardQosPolicies.rst:491]: (typo) Follow the same structure as the example below

```suggestion
  :ref:`historyqospolicy` |HistoryQosPolicy::kind-api| being set to |KEEP_ALL_HISTORY_QOS-api|, when the resource limits are reached, the behavior
```

File: docs/fastdds/dds_layer/core/policy/standardQosPolicies.rst [hunk 2]
```diff
@@ -478,24 +478,26 @@ The HistoryQos must be set consistently with the :ref:`resourcelimitsqospolicy`,
...
-* In case the :ref:`reliabilityqospolicy` is set as |RELIABLE_RELIABILITY_QOS-api|, the |HistoryQosPolicy::kind-api| is
-  configured as |KEEP_ALL_HISTORY_QOS-api|, the resource limits are reached, and based on the :ref:`durabilityqospolicy`
-  configuration:
-
-  * If the |DurabilityQosPolicy::kind-api| is configured as |VOLATILE_DURABILITY_QOS-api|, the DataWriter publication
-    will discard the oldest samples.
-    Those samples may belong to different :ref:`instances<dds_layer_topic_instances>`.
-  * If the |DurabilityQosPolicy::kind-api| is configured as |TRANSIENT_LOCAL_DURABILITY_QOS-api| or
-    |TRANSIENT_DURABILITY_QOS-api|, the DataWriter publication will be blocked.
+* In the case of the :ref:`reliabilityqospolicy` being set to |RELIABLE_RELIABILITY_QOS-api| and the
```
Comment (JesusPoderoso) [inline docs/fastdds/dds_layer/core/policy/standardQosPolicies.rst:490]: Follow the same structure as the example below

```suggestion
* In the case of the :ref:`reliabilityqospolicy` |ReliabilityQosPolicyKind-api| being set to |RELIABLE_RELIABILITY_QOS-api| and the
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]

**Review by JesusPoderoso — APPROVED** [review]
LGTM

**Comment by EduPonz** [comment]
@richiprosima please test this

**Comment by JesusPoderoso** [comment]
@Mergifyio backport 2.10.x

**Pull request merged**